### DISCUSSION TITLE
Local Action not found in workflow

### DISCUSSION THREAD

**Discussion by bradmartin** [community/community#25153] [Actions]
<p>I have my main workflow (/.github/workflows/builds.yml) :</p>
<pre><code>jobs:
  job1:
    name: PT.M Android Build
    runs-on: macos-latest
    steps:
      - name: Clone Repo
        uses: actions/checkout@v1

      - name: Shared Setup
        uses: ./.github/actions/shared-setup/shared-setup
</code></pre>
<p>Attempting to use a local action that runs other actions that end up being duplicated across other jobs in the same workflow. I’m trying to avoid duplicating a lot of the same process in my CI Builds by having a local action to call that runs the same steps needed when executed.</p>
<p>I’ve simplified my local action to this (/.github/actions/shared-setup.yml) :</p>
<pre><code>- name: Setup SSH-Agent
  uses: webfactory/ssh-agent@v0.2.0
  with:
    ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}

- name: Setup JDK 1.8
  uses: actions/setup-java@v1
  with:
    java-version: 1.8
</code></pre>
<p>When I run the checks the following error happens:</p>
<pre><code>##[error]Can't find 'action.yml' or 'Dockerfile' under '/Users/runner/runners/2.163.1/work/permobil-client/permobil-client/.github/actions/shared-setup'. Did you forget to run actions/checkout before running your local action?
</code></pre>
<p>I stumbled on this topic <a href="https://github.community/t/path-to-action-in-the-same-repository-as-workflow/16952">https://github.community/t5/GitHub-Actions/Path-to-action-in-the-same-repository-as-workflow/td-p/38761</a> but from the answer and the docs I assume I have everything configured correctly. I’m running checkout prior to running my local action.</p>
<p>Any ideas?</p>

**Comment by bradmartin** [comment]
<p>I did successfully find the action now, however, maybe this approach is wrong for what I’m doing.</p>
<ol>
<li>
<p>an action needs the action.yml metadata file</p>
</li>
<li>
<p>My uses statement would be </p>
<ul>
<li>name: Shared Setup<br>
uses: ./.github/actions/shared-setup</li>
</ul>
</li>
</ol>
<p>So now my workflow does find the action but another issue is introduced:</p>
<p> My main workflow runs on <code>macos-latest</code> and I’m unsure how to run a local action with this workflow.</p>
<p>It’s possible I’m trying to use a local action in a way that is not supported. At the end of the day I’m just trying to reduce duplication across jobs. I was unable to get YAML aliases working with github actions (assumed not supported) so I thought maybe using a local action would help keep my workflow DRY. Instead of repeating the same setup steps in 4 jobs.</p>

**Comment by BrightRan** [comment, marked as answer]
<p>Your request is similar to set up job/step templates which can be referenced in workflow definitions. But as I have mentioned in <a href="https://github.community/t/can-i-use-action-as-a-full-job-to-encapsulate-all-ci-process/17238">this ticket</a>, currently GitHub Actions does not support job/step templates.</p>
<p>Maybe you can try to encapsulate the repeated steps into one action, for example if the repeated steps are executing some command lines, so that you just need to use this action in each job. However, most of the time, it is very hard and troublesome to encapsulate multiple jobs/steps into one action.</p>
<p>I recommend that you can directly report a feature request <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a>. That will allow you to directly interact with the appropriate engineering team, and make it more convenient for the engineering team to collect and categorize your suggestions.</p>

**Comment by zackify** [comment]
<p>I think all of us would love this feature. Just today the team I’m on asked if we can reuse this instead of copy pasting for minor differences in workflow.</p>

**Comment by rohitattaluri** [comment]
<p>Do we need to crate a folder named <strong>actions</strong> parallel to <strong>workflow</strong> folder here? and if so, what should actions folder consists of?</p>
<p>below is my .yml file</p>
<pre><code>name: Dart CI

on:
  push:
    branches: [feature/tests]
  pull_request:
    branches: [feature/tests]

jobs:
  test:
    name: Test on ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, windows-latest, macos-latest]
    steps:
      - uses: actions/checkout@v1
      - uses: actions/setup-java@v1
        with:
          java-version: '12.x'

      # Integration test for home page empty state
      #- uses: what is the path I should write here.
        with:
          token: '${{ secrets.GITHUB_TOKEN }}'
      - run: flutter drive --target=test_driver/listing_details/home_tab/home_tab_empty_state.dart
</code></pre>
<p>Highlighted circle is the hierarchy of the action files in my repo.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/b758898de5b896901f1e943e76ed011755df5681.png" data-download-href="/uploads/short-url/q9X4bc6caB1BZWEzXahIK0UTAQh.png?dl=1" title="Annotation 2020-04-02 184109.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/59006955/181089170-74c034f7-5a4a-4641-8ef8-2495dfb9c3c3.png" alt="Annotation 2020-04-02 184109.png" data-base62-sha1="q9X4bc6caB1BZWEzXahIK0UTAQh" width="690" height="355" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/b758898de5b896901f1e943e76ed011755df5681_2_690x355.png, https://aws1.discourse-cdn.com/github/optimized/1X/b758898de5b896901f1e943e76ed011755df5681_2_1035x532.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/b758898de5b896901f1e943e76ed011755df5681_2_1380x710.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/b758898de5b896901f1e943e76ed011755df5681_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Annotation 2020-04-02 184109.png</span><span class="informations">1388×715 61.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
